contracts/Fees.sol

```diff
@@ -28,43 +29,49 @@ abstract contract Fees is IFees, Owned {
         controllerGasLimit = _controllerGasLimit;
     }
 
-    function _fetchProtocolFees(PoolKey memory key)
-        internal
-        view
-        returns (uint8 protocolSwapFee, uint8 protocolWithdrawFee)
-    {
+    function _fetchProtocolFees(PoolKey memory key) internal view returns (uint24 protocolFees) {
+        uint16 protocolSwapFee;
+        uint16 protocolWithdrawFee;
         if (address(protocolFeeController) != address(0)) {
             // note that EIP-150 mandates that calls requesting more than 63/64ths of remaining gas
             // will be allotted no more than this amount, so controllerGasLimit must be set with this
             // in mind.
             if (gasleft() < controllerGasLimit) revert ProtocolFeeCannotBeFetched();
             try protocolFeeController.protocolFeesForPool{gas: controllerGasLimit}(key) returns (
-                uint8 updatedProtocolSwapFee, uint8 updatedProtocolWithdrawFee
+                uint16 updatedProtocolSwapFee, uint16 updatedProtocolWithdrawFee
             ) {
                 protocolSwapFee = updatedProtocolSwapFee;
                 protocolWithdrawFee = updatedProtocolWithdrawFee;
-            } catch {}
 
+                if (protocolSwapFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(protocolSwapFee);
+                if (protocolWithdrawFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(protocolWithdrawFee);
+
+                protocolFees = (uint24(protocolSwapFee) << 12) | protocolWithdrawFee;
+            } catch {}
             _checkProtocolFee(protocolSwapFee);
             _checkProtocolFee(protocolWithdrawFee);
         }
     }
 
     /// @notice There is no cap on the hook fee, but it is specified as a percentage taken on the amount after the protocol fee is applied, if there is a protocol fee.
-    function _fetchHookFees(PoolKey memory key) internal view returns (uint8 hookSwapFee, uint8 hookWithdrawFee) {
+    function _fetchHookFees(PoolKey memory key) internal view returns (uint24 hookFees) {
         if (key.fee.hasHookSwapFee()) {
-            hookSwapFee = IHookFeeManager(address(key.hooks)).getHookSwapFee(key);
+            uint16 swapFee = IHookFeeManager(address(key.hooks)).getHookSwapFee(key);
+            if (swapFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(swapFee);
+            hookFees |= (uint24(swapFee) << 12);
         }
 
         if (key.fee.hasHookWithdrawFee()) {
-            hookWithdrawFee = IHookFeeManager(address(key.hooks)).getHookWithdrawFee(key);
+            uint16 withdrawFee = IHookFeeManager(address(key.hooks)).getHookWithdrawFee(key);
+            if (withdrawFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(withdrawFee);
+            hookFees |= withdrawFee;
         }
     }
 
-    function _checkProtocolFee(uint8 fee) internal pure {
+    function _checkProtocolFee(uint16 fee) internal pure {
```

Comment:
Can we leave an @dev comment here saying that only 12 least significant bits are used even though this is a uint16.